### PR TITLE
misc fixes: cancel throw exception, camera does not save properly

### PR DESCRIPTION
**Pull request by maxatcn**
Misc Fixes:
1. gallery cancel throw exception
2. camera could not save image properly
3. camera cancel throw exception
4. after cancel, could not launch camera again.

can you please review and merge?  Thank you.

### CODE REVIEW

**Comment by maxatcn** [comment]
I signed it!

File: packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerPlugin.java [hunk 1]
```diff
@@ -113,6 +116,13 @@ public void onImageReady(List<Image> images) {
               }
             });
         return true;
+      } else if (resultCode == Activity.RESULT_CANCELED) {
```
Comment (Alexays) [inline packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerPlugin.java]: You could just do this
```
} else if (resultCode != Activity.RESULT_CANCELED) {
       pendingResult.error("PICK_ERROR", "Error taking photo", null);
}
pendingResult = null;
methodCall = null;
```

Comment (maxatcn) [reply packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerPlugin.java]: Good point. Thanks for the comment @Alexays .  I updated the code and can you please review again?

**Comment by Hixie** [comment]
cc @mit-mit

**Comment by mit-mit** [comment]
cc @mravn-google

**Pull request merged**